### PR TITLE
Update draggable regions when changing BrowserView

### LINKED ISSUE #12150: -webkit-app-region: drag prevents clicks on overlapping browser view
* Electron version: 1.8.2
* Operating system: macos 10.13.3

Reproducing repo: https://github.com/ibash/electron-webkit-app-drag

Demonstrates a bug where `-webkit-app-region: drag` causes a browser view in front of it to miss click events. Oddly - it does not prevent hover events.

Steps:
1. Clone https://github.com/ibash/electron-webkit-app-drag, `yarn install`, `yarn start`
2. Click "button1"
3. Click "button2"

Expected:

Clicking button1 or button2 pops up an alert: "button clicked!"

Actual:

Only clicking button2 pops up an alert: "button clicked!"

### PR DESCRIPTION
**Pull request by poiru**
This PR is best reviewed by looking at each individual commit.

This should be cherry-picked to `2-0-x`, but it will require a separate PR due to conflicts.

Fixes #12150.

### CODE REVIEW

File: atom/browser/native_browser_view.cc [hunk 1]
```diff
@@ -7,14 +7,23 @@
...
-    brightray::InspectableWebContentsView* web_contents_view)
-    : web_contents_view_(web_contents_view) {}
+    brightray::InspectableWebContents* inspectable_web_contents)
+    : inspectable_web_contents_(inspectable_web_contents) {}
 
 NativeBrowserView::~NativeBrowserView() {}
 
+brightray::InspectableWebContentsView*
+NativeBrowserView::GetInspectableWebContentsView() {
+  return inspectable_web_contents_->GetView();
```
Comment (deepak1556) [inline atom/browser/native_browser_view.cc]: Can you check the availability of `inspectable_web_contents_` before using it. Its possible for the `webContents` to get destroyed before the view.

File: atom/browser/api/atom_api_browser_window_mac.mm [hunk 2]
```diff
@@ -80,30 +80,24 @@ - (NSView*)hitTest:(NSPoint)aPoint {
...
-    window_->browser_view()->UpdateDraggableRegions(system_drag_exclude_areas);
+    window_->browser_view()->UpdateDraggableRegions(drag_exclude_rects);
 
   // Create and add a ControlRegionView for each region that needs to be
   // excluded from the dragging.
-  for (std::vector<gfx::Rect>::const_iterator iter =
-           system_drag_exclude_areas.begin();
-       iter != system_drag_exclude_areas.end();
-       ++iter) {
+  for (const auto& rect : drag_exclude_rects) {
```
Comment (felixrieseberg) [inline atom/browser/api/atom_api_browser_window_mac.mm]: I love how much simpler this is ✨

File: atom/browser/native_browser_view.cc (hunk 1 above)
Comment (poiru) [reply atom/browser/native_browser_view.cc]: I think the existing code already had similar issues. I'll work on fixing it in a separate PR, but for the purposes of fixing the draggable region bug, I think we should go ahead and merge this as-is.

**Comment by ckerr** [comment]
I don't understand why this was merged with new possible nullptr dereferences, especially since we intend to backport this to other branches. Saying "let's merge this as-is and fix those in another PR" just means we now have two branches that will need manual backporting

**Comment by jkleinsc** [comment]
@ckerr I merged it because folks approved it.  If it wasn't ready to go, I would prefer folks ask for changes vs approving.

**Comment by poiru** [comment]
@ckerr The old code would have crashed as well with a different call stack. I agree that it should be fixed, but I don't think that bit necessarily needs to be backported.

**Comment by ckerr** [comment]
@jkleinsc that's fair, I should have said "I don't understand why this is approved". Completely agree that approved code passing CI should be merged without being second-guessed. :+1:

@poiru I agree with you about the old code being unsafe too, we're on the same page there. My only point is I don't see why we need a separate PR for the nullptr guards. Is there more safety cleanup to be done in addition to that, and you want to group it all together?

**Comment by ckerr** [comment]
@poiru also, nice job getting the backports PRed so quickly :smile_cat:

**Comment by poiru** [comment]
@ckerr Ah, I see! Unfortunately it's not as simple as adding `nullptr` checks, we need to refactor some code to make it safe.

**Pull request merged**